### DISCUSSION TITLE
Support http auth for health checks

### DISCUSSION THREAD

**Discussion by HLFH** [basecamp/kamal-proxy#100] [Ideas and Issue Triage]
It would be great to support two parameters:
 - `health-check-username`
 - `health-check-password`
 
Example of Docker app that needs HTTP auth: Transmission.
https://hub.docker.com/r/linuxserver/transmission

`kamal-proxy deploy` does not work when the root URL requires HTTP auth.

**Comment by zdehasek** [comment]
Yes, this feature would be really cool.
I’m currently deploing Prometheus using HTTP authentication, and it would be great if Kamal Proxy could handle this. Otherwise, I won’t be able to deploy it at all.

We don’t even need new parameters: HTTP auth could be set like this: user:password@example.org. So, if the proxy allows providing a full URL instead of just the path, that would be perfect.

I am not a go developer, but if PR like this will be merged, I am happy to do my best :)

**Comment by cavi21** [comment]
Hi, just joining the conversation in in order to know if there is interest to have this supported by kamal-proxy? maybe @kevinmcconnell, whenever you find the time, if can share your ideas or concerns about this it would be great :)!

**Comment by monorkin** [comment]
I'm curious, why would you want to protect the health check endpoint with basic auth?

**Reply by cavi21** [reply]
Some services that could be deployed as accessories with Kamal that has `proxy` config could have this requirement, for example `elastic/elasticsearch`, but there are others too, there is also a conversation (https://github.com/basecamp/kamal/issues/1122) that touch this case

**Reply by cavi21** [reply]
Hi @monorkin did you had the chance to check this? maybe another option will be to be able to define the http status code to consider healthy, so maybe the **200** can be overridden with **403** for the example I shared
